apps/meteor/server/methods/setUserActiveStatus.ts

```diff
@@ -17,20 +20,40 @@ Meteor.methods<ServerMethods>({
 		check(userId, String);
 		check(active, Boolean);
 
-		if (!Meteor.userId()) {
+		const uid = Meteor.userId();
+
+		if (!uid) {
 			throw new Meteor.Error('error-invalid-user', 'Invalid user', {
 				method: 'setUserActiveStatus',
 			});
 		}
 
-		const uid = Meteor.userId();
-
 		if (!uid || (await hasPermissionAsync(uid, 'edit-other-user-active-status')) !== true) {
 			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
 				method: 'setUserActiveStatus',
 			});
 		}
 
+		const { federated } = (await Users.findOneById<Pick<IUser, 'federated'>>(uid, { projection: { federated: 1 } })) || {};
+
+		if (federated) {
+			throw new Meteor.Error('error-not-allowed', 'Deactivating federated user is not allowed', { method: 'setUserActiveStatus' });
+		}
+
+		const remoteUser = await MatrixBridgedUser.getExternalUserIdByLocalUserId(userId);
+
+		if (remoteUser) {
+			if (active) {
```

Comment:
Is this correct? If so, it seems the error message is wrong.

Reply:
`active` here is from the payload, so "mark this user active again". 

https://github.com/RocketChat/Rocket.Chat/blob/8fc6ca8b4e09b38ea9c74dbed586b8bc3694cf09/apps/meteor/app/api/server/v1/users.ts#L104-L111